src/unix/terminal.rs

```diff
@@ -395,8 +395,14 @@ struct Winsize {
 fn get_winsize(fd: c_int) -> io::Result<Winsize> {
     let mut winsz: Winsize = unsafe { zeroed() };
 
+    #[cfg(not(target_os = "freebsd"))]
     let res = unsafe { ioctl(fd, TIOCGWINSZ, &mut winsz) };
 
+    // NOTE: this is a temporary fix to a libc bug described in:
+    //  https://github.com/rust-lang/libc/pull/704
+    #[cfg(target_os = "freebsd")]
+    let res = unsafe { ioctl(fd, TIOCGWINSZ as u64, &mut winsz) };
```

Comment:
A couple minor points: I believe attributes on expressions landed in stable rustc only a few days ago, so I think it would be better to implement this as `cfg`-marked functions, to continue supporting the old stable. Also, I think the cast would be better done as `as c_ulong`.